fastapi_filter/contrib/sqlalchemy/filter.py

```diff
@@ -86,8 +87,15 @@ def filter(self, query: Union[Query, Select]):
                 else:
                     operator = "__eq__"
 
-                model_field = getattr(self.Constants.model, field_name)
-                query = query.filter(getattr(model_field, operator)(value))
+                if field_name == self.Constants.search_param and hasattr(self.Constants, "search_fields"):
+
+                    def search_filter(field):
+                        return func.lower(getattr(self.Constants.model, field)).like("%" + value.lower() + "%")
```

Comment:
My goal was to compare similar strings. the `func.lower` will convert the string of the columns to lower case in SQL, and `value.lower()` will convert the search string to lower case in Python.

Since I am using Postgres which `LIKE` is case sensitive, this is necessary to have case insensitive searches. If the DB engine has case-insensitive `LIKE`, I can see how this feels like overkill.

I spent a little bit of time in the sqlalchemy docs, and found that it [implements `ILIKE`](https://docs.sqlalchemy.org/en/14/core/sqlelement.html#sqlalchemy.sql.expression.ColumnOperators.ilike), even for db engines that don't support it natively. I think this is a better solution. I changed the code to use this.

Let me know what you think.